diff --git a/src/Element/EntityBrowserElement.php b/src/Element/EntityBrowserElement.php
index 2b1c572..de13094 100644
--- a/src/Element/EntityBrowserElement.php
+++ b/src/Element/EntityBrowserElement.php
@@ -164,38 +164,49 @@ class EntityBrowserElement extends FormElement {
       ['cardinality' => ['cardinality' => $element['#cardinality']]]
     );
 
-    $display = $entity_browser->getDisplay();
-    $display->setUuid(sha1(implode('-', array_merge([$complete_form['#build_id']], $element['#parents']))));
-    $element['entity_browser'] = [
-      '#eb_parents' => array_merge($element['#parents'], ['entity_browser']),
-    ];
-    $element['entity_browser'] = $display->displayEntityBrowser(
-      $element['entity_browser'],
-      $form_state,
-      $complete_form,
-      [
-        'validators' => $validators,
-        'selected_entities' => $entity_browser_preselected_entities,
-        'widget_context' => $element['#widget_context'],
-      ]
-    );
+    // Error if entity_browser not found
+    // ex: entity_browser has been deleted
+    if (!$entity_browser) {
+      $element['entity_browser'] = array(
+        '#type' => 'markup',
+        '#markup' => t('Entity browser not found'),
+      );
+    }
+    // Display entity_browser
+    else {
+      $display = $entity_browser->getDisplay();
+      $display->setUuid(sha1(implode('-', array_merge([$complete_form['#build_id']], $element['#parents']))));
+      $element['entity_browser'] = [
+        '#eb_parents' => array_merge($element['#parents'], ['entity_browser']),
+      ];
+      $element['entity_browser'] = $display->displayEntityBrowser(
+        $element['entity_browser'],
+        $form_state,
+        $complete_form,
+        [
+          'validators' => $validators,
+          'selected_entities' => $entity_browser_preselected_entities,
+          'widget_context' => $element['#widget_context'],
+        ]
+      );
 
-    $hidden_id = Html::getUniqueId($element['#id'] . '-target');
-    $element['entity_ids'] = [
-      '#type' => 'hidden',
-      '#id' => $hidden_id,
-      // We need to repeat ID here as it is otherwise skipped when rendering.
-      '#attributes' => ['id' => $hidden_id, 'class' => ['eb-target']],
-      '#default_value' => $default_value,
-    ];
+      $hidden_id = Html::getUniqueId($element['#id'] . '-target');
+      $element['entity_ids'] = [
+        '#type' => 'hidden',
+        '#id' => $hidden_id,
+        // We need to repeat ID here as it is otherwise skipped when rendering.
+        '#attributes' => ['id' => $hidden_id, 'class' => ['eb-target']],
+        '#default_value' => $default_value,
+      ];
 
-    $element['#attached']['drupalSettings']['entity_browser'] = [
-      $entity_browser->getDisplay()->getUuid() => [
-        'cardinality' => $element['#cardinality'],
-        'selection_mode' => $element['#selection_mode'],
-        'selector' => '#' . $hidden_id,
-      ],
-    ];
+      $element['#attached']['drupalSettings']['entity_browser'] = [
+        $entity_browser->getDisplay()->getUuid() => [
+          'cardinality' => $element['#cardinality'],
+          'selection_mode' => $element['#selection_mode'],
+          'selector' => '#' . $hidden_id,
+        ],
+      ];
+    }
 
     return $element;
   }
